github-dark.user.css

```diff
@@ -189,6 +189,35 @@
     display: table-cell !important;
   } EOT;
 }
+@advanced dropdown custom-chrome-scrollbars "Custom Chrome Scrollbars" {
+  disable "Disable" <<<EOT
+  EOT;
+  enable "Enable" <<<EOT
```

Comment:
Will this enable or disable it by default? If the issues related to it are not too severe, I'd lean towards having it enabled by default.

Reply:
They dont work in chrome too well here.

Reply:
To my knowledge, they work flawlessly in Chrome, we had them for years. Evidence so far show some buggy behaviour in Firefox (due to being a new feature) and exotic browsers. No need to disable by default because of that.

Reply:
The old CSS yes, but the new CSS no. Unless Im missing something, some testing across chrome based browsers shows problems.

The problem is this

```css
  @supports (scrollbar-color: rgba(140, 140, 140, .5) transparent) and (scrollbar-width: thin) {
    * {
      scrollbar-color: rgba(140, 140, 140, .5) transparent !important;
      scrollbar-width: thin !important;
    }
  }
  @supports (not (scrollbar-color: rgba(140, 140, 140, .5) transparent)) and (not (scrollbar-width: thin)) {
    ::-webkit-scrollbar {
      max-width: 5px !important;
      max-height: 5px !important;
      background: #1d1d1d !important;
    }
    ::-webkit-scrollbar-track, ::-webkit-scrollbar-corner {
      background: transparent !important;
    }
    ::-webkit-scrollbar-thumb {
      background: rgba(140, 140, 140, .5) !important;
    }
    ::-webkit-scrollbar-thumb:hover {
      background: rgba(79, 140, 201, .8) !important;
      background: rgba(/*[[base-color-rgb]]*/, .8) !important;
    }
  }
```

Like you say, this CSS below There was no problem. 

```css
    ::-webkit-scrollbar {
      max-width: 5px !important;
      max-height: 5px !important;
      background: #1d1d1d !important;
    }
    ::-webkit-scrollbar-track, ::-webkit-scrollbar-corner {
      background: transparent !important;
    }
    ::-webkit-scrollbar-thumb {
      background: rgba(140, 140, 140, .5) !important;
    }
    ::-webkit-scrollbar-thumb:hover {
      background: rgba(79, 140, 201, .8) !important;
      background: rgba(/*[[base-color-rgb]]*/, .8) !important;
    }
```

Reply:
I've had no issues with the updated scrollbar styles in chrome on windows over the last month. I'm on chrome 71 and it's using `::-webkit-scrollbar`.

Reply:
Its unreliable, see this

```css
/* ==UserStyle==
@name         scrollbars
@version      0.0.1
@description  scrollbars test
@namespace    something
@advanced color custom-scrollbar-color "Scrollbar color" rgba(140, 140, 140, .5)"
@advanced color custom-track-color "Track color" "transparent"
==/UserStyle== */

@supports (scrollbar-color: rgba(140, 140, 140, .5) transparent) and (scrollbar-width: thin) {
  * {
  scrollbar-color: /*[[custom-scrollbar-color]]*/ /*[[custom-track-color]]*/ !important;
  scrollbar-width: thin !important;
  }
}
@supports (not (scrollbar-color: rgba(140, 140, 140, .5) transparent)) and (not (scrollbar-width: thin)) {
  ::-webkit-scrollbar {
    max-width: 5px !important;
    max-height: 5px !important;
    background: /*[[custom-track-color]]*/ !important;
  }
  ::-webkit-scrollbar-track, ::-webkit-scrollbar-corner {
    background: /*[[custom-track-color]]*/ !important;
  }
  ::-webkit-scrollbar-thumb {
    background: rgba(/*[[custom-scrollbar-color]]*/, .5) !important;
  }
  ::-webkit-scrollbar-thumb:hover {
    background: rgba(79, 140, 201, .8) !important;
    background: rgba(/*[[custom-scrollbar-color]]*/, .8) !important;
  }
}
```

Task, set a scrollbar color and when done change it.

result

![capture](https://user-images.githubusercontent.com/31389848/51303359-6749f600-1a35-11e9-9a70-e1affc41ee91.PNG)

Fail.

Same happens with the current code, you cant change the color, other unreliable behavior depends on browser, doesnt always behave like expected.

This is opinionated and doesnt belong in GHD and if it stays should be optional OFF as default because its opinionated,

Reply:
## Bottom line 

This is opinionated and doesnt really belong in GHD and if it stays should be optional OFF as default because its opinionated,

You have #820 and Ive had more than one person at discord Stylus channel complain that when they use GitHub Dark their chrome scrollbars are broken or thin.

Whats the problem with making this optional? I dont get it.

Reply:
I'm not a huge fan of the 5px width/height for scrollbars. Adding a UserCSS setting for custom values would be much better.

Dark scrollbars on the hand are a must in my opinion. It'd be a bit silly to leave them unchanged.

Reply:
im not against the scrollbar being styled i just dont like it and think it should be optional. especially so it doesnt conflict with my scrollbar styling. same goes for style background patterns. should be optional.

Reply:
Ill make the new style, I have some ideas how to handle some of the weirdness, but dont close this, Ill update and push when Im ready, likely in the morning ;)

Reply:
OK see https://github.com/StylishThemes/Overlay-Scrollbars

Will push this into master and make a release shortly.

Thx you for your patience.

Reply:
Why's the new style apply to all sites instead of just github?

Reply:
I'd say make two styles in that repo. One for GitHub only, the other is global.

Reply:
thumbs up

Reply:
@xt0rted https://github.com/StylishThemes/Overlay-Scrollbars/blob/master/github-overlay-scrollbars.user.css

Theres a global also for whoever like me wants it.